libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1605,48 +1604,33 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 				if (!dynamic_cast<RationalNumberType const&>(*types[i]).mobileType())
 					m_errorReporter.fatalTypeError(3390_error, components[i]->location(), "Invalid rational number.");
 
-			if (_tuple.isInlineArray())
+			if (_tuple.isInlineArray() and types[i]->category() != Type::Category::InlineArray)
```

Comment:
Wow! I did not know that it is possible to use `and` - I always used `&&`. 😅

Reply:
The funny thing is, I tried to use it when I first joined too. Then I switched to `&&` and `||` for the sake of consistency :)